Library/Homebrew/cmd/outdated.rb

```diff
@@ -22,43 +24,66 @@ def outdated_args
       flag   "--json",
              description: "Print output in JSON format. Currently the default and only accepted "\
                           "value for <version> is `v1`. See the docs for examples of using the JSON "\
-                          "output: <https://docs.brew.sh/Querying-Brew>"
+                          "output: <https://docs.brew.sh/Querying-Brew>. "\
+                          "By default, this option treats all arguments as formulae. "\
+                          "To treat arguments as casks, use the --cask-only option."
       switch "--fetch-HEAD",
              description: "Fetch the upstream repository to detect if the HEAD installation of the "\
                           "formula is outdated. Otherwise, the repository's HEAD will only be checked for "\
                           "updates when a new stable or development version has been released."
+      switch "--greedy",
+             description: "Print outdated casks with `auto_updates` or `version :latest`"
+      switch "--formula",
+             description: "Treat all arguments as formulae"
+      switch "--cask",
+             description: "Treat all arguments as casks"
       switch :debug
       conflicts "--quiet", "--verbose", "--json"
+      conflicts "--formula-only", "--cask-only"
     end
   end
 
   def outdated
     outdated_args.parse
 
-    formulae = if args.resolved_formulae.blank?
-      Formula.installed
-    else
-      args.resolved_formulae
-    end
+    formula_only = args.formula?
     if args.json
       raise UsageError, "invalid JSON version: #{args.json}" unless ["v1", true].include? args.json
 
-      outdated = print_outdated_json(formulae)
+      # When user asks for json, print json for formula only unless the user explicitly asks for casks
+      formula_only = !args.cask?
+    end
+
+    if formula_only
```

Comment:
```suggestion
    outdated = if formula_only
```

then you can remove the `outdated` assignments